doc/policy/version3_transactions.md

```diff
@@ -0,0 +1,100 @@
+transaction's incentive compatibility. Imagine the original transaction, A, has a child B and
+co-parent C (i.e. B spends from A and C). C also has another child, D. B is one of the original
+transactions and thus its ancestor feerate must be lower than the package's. However, this may be an
+underestimation because D can bump C without B's help. This is resolved if V3 transactions can only
+have V3 ancestors, as then C cannot have another child.
+
+4. A V3 transaction cannot have more than 1 unconfirmed descendant.
+
+*Rationale*: (upper bound) the larger the descendant limit, the more transactions may need to be
+replaced. This is a problematic pinning attack, i.e., a malicious counterparty prevents the
+transaction from being replaced by adding many descendant transactions that aren't fee-bumping.
+See example #1 in ["Rule 3" Pining](#Rule-3-Pinning) section above.
+
+*Rationale*: (lower bound) at least 1 descendant is required to allow CPFP of the presigned
+transaction. The contract protocol can create presigned transactions paying 0 fees and 1 output for
+attaching a CPFP at broadcast time ("anchor output"). Without package RBF, multiple anchor outputs
+would be required to allow each counterparty to fee-bump any presigned transaction. With package
+RBF, since the presigned transactions can replace each other, 1 anchor output is sufficient.
+
+5. A V3 transaction that has an unconfirmed V3 ancestor cannot be larger than 1000 virtual bytes.
+
+*Rationale*: (upper bound) the larger the descendant size limit, the more vbytes may need to be
+replaced. With default limits, if the child is e.g. 100,000vB, that might be an additional
+100,000sats (at 1sat/vbyte) or more, depending on the feerate. Restricting all children to 1000vB
+bounds the potential fees by 1/100.
+
+*Rationale*: (lower bound) the smaller this limit, the fewer UTXOs a child may use to fund this
+fee-bump. For example, only allowing the V3 child to have 2 inputs would require L2 protocols to
+manage a wallet with high-value UTXOs and make batched fee-bumping impossible. However, as the
+fee-bumping child only needs to fund fees (as opposed to payments), just a few UTXOs should suffice.
+
+*Rationale*: With a limit of 1000 virtual bytes, depending on the output types, the child can have
+6-15 UTXOs, which should be enough to fund a fee-bump without requiring a carefully-managed UTXO
+pool. With 1000 virtual bytes as the descendant limit, the cost to replace a V3 transaction has much
+lower variance.
+
+*Rationale*: This makes the rule very easily "tacked on" to existing logic for policy and wallets.
+A transaction may be up to 100KvB on its own (`MAX_STANDARD_TX_WEIGHT`) and 101KvB with descendants
+(`DEFAULT_DESCENDANT_SIZE_LIMIT_KVB`). If an existing V3 transaction in the mempool is 100KvB, its
+descendant can only be 1000vB, even if the policy is 10KvB.
+
+4. A V3 transaction cannot have more than 1 unconfirmed ancestor.
```

Comment:
Also, I think this could be a minor issue for LN implementation. If some of your counterparties have not upgraded their software to support V3 transactions you might have a subset of your channel transaction still under V2 during the transition period. If you have a common child for batched fee-bumping shared between a V2 commitment transaction and V3 commitment transaction, the whole operation is likely to fail. Of course, batch fee-bumping should be considered harmful for time-sensitive commitment transactions (i.e ones with HTLC outputs) though we might have further silent issues even when it's not time-sensitive. In my opinion, this is a potential issue LN devs should be aware of, processing between V2 and V3 channels should be well isolated.